gap/PolygonalComplexes/modification.gi

```diff
@@ -2016,6 +2018,49 @@ InstallMethod( TetrahedralExtension,
 	return SimplicialSurfaceByDownwardIncidence(VOE,EOF);
 end
 );
+
+InstallMethod( TetrahedralReduction,
+    "for a simplicial surface and a vertex",
+    [IsSimplicialSurface, IsPosInt],
+    function(surface, vertex)
+	local f,eof,voe,i,L,ind,faces,remElm,edges,verts,g;
+	remElm:=function(L,ind)
+		local i,res;
+		res:=[];
+		for i in [1..Length(L)] do
+			if IsBound(L[i]) and i <> ind then 
+				res[i]:=L[i];
+			fi;
+		od; 
+		return res;
+	end;
+
+	eof:=ShallowCopy(EdgesOfFaces(surface));
+	voe:=ShallowCopy(VerticesOfEdges(surface));
+	faces:=FacesOfVertex(surface,vertex);
+	if Length(faces)=3 and IsInnerVertex(surface,vertex) then
+		for f in faces do
+			eof:=remElm(eof,f);
```

Comment:
Why don't you use Remove(eof,f)?

Reply:
Remove() returns a dense list, which will mix up the numbering of the faces. In this line I only want to remove the edges of face f which will give me a list that is no more dense, but all the other face numbers remain the same.